nemo/collections/nlp/data/machine_translation/one_side_dataset.py

```diff
@@ -267,7 +267,20 @@ def _build_sample(self, fname):
         return ids, mask
 
     def __iter__(self):
-        return self._dataset.__iter__()
+        # We need to wrap an infinite generator since the actual files
+        # within the tar files contains large chunks of contiguous data.
```

Comment:
Each pickle file within a tar file contains exactly one minibatch for the sentence-level `one_side_dataset`. So I'm not sure why you need this infinite generator

Reply:
Agree